modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -407,6 +415,10 @@ private void handleSafeTimeSyncCommand(SafeTimeSyncCommand cmd, long commandInde
 
         // We MUST bump information about last updated index+term.
         // See a comment in #onWrite() for explanation.
+        advanceLastAppliedIndexConsistently(commandIndex, commandTerm);
```

Comment:
Seems that we also miss lastAppliedIndexUpdate in case of `handleFinishTxCommand`.

Reply:
We don't, the index gets updated by the `compareAndSet()` method internally. I wrote a test for that.

Reply:
Yep, now I can. Missed that.